### PR TITLE
Add tests for TrustyAI Notebooks

### PR DESCRIPTION
**Pull request by dibryant**
Fixes for #182 
## Description

Run Jupyter TrustyAI Notebooks for both the versions of Python, i.e 3.8 and 3.9. Run the test notebooks in the relevant environment. The tests should execute with all passing.
## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    "        nobias = pd.read_csv(url_unbiased, index_col=False)\n",
+    "        \n",
+    "        nobias = pd.read_csv(url_unbiased, index_col=False)\n",
+    "        nobias.groupby(['gender', 'income'])['income'].count()\n",
+    "        nobias.groupby(['gender', 'income'])['income'].count().unstack().plot.bar()\n",
+    "        \n",
+    "        nobias_privileged = nobias[nobias.gender == 1]\n",
+    "        nobias_unprivileged = nobias[nobias.gender == 0]\n",
+    "        favorable = output(\"income\", dtype=\"number\", value=1)\n",
+    "        score = statistical_parity_difference(privileged=nobias_privileged,\n",
```
Comment (atheo89) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb:86]: I would suggest using a check here to evaluate the result of the score like:
```
self.assertTrue(score >= 0.0036255104824703954)  
        print("On the test_fairness_metrics test case the statistical_parity_difference score for this dataset is between the threshold [-0.1,0.1], which classifies the model as reasonably fair.")
```

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    "\n",
+    "    def test_python_version(self): \n",
+    "        expected_major_minor = '3.9' # Set the expected version (x.y) \n",
+    "        actual_major_minor = '.'.join(python_version().split('.')[:2])\n",
+    "        self.assertEqual(actual_major_minor, expected_major_minor, \"incorrect version\")\n",
+    "\n",
+    "    def test_trustyai_version(self): \n",
+    "        expected_major_minor = '0.2' # Set the expected version (x.y) \n",
+    "        actual_major_minor = '.'.join(trustyai.__version__.split('.')[:2]) \n",
+    "        self.assertEqual(actual_major_minor, expected_major_minor, \"incorrect version\")\n",
```
Comment (atheo89) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb:74]: Needs a single line here

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb [hunk 3]
```diff
@@ -0,0 +1,133 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "df03ef8f-cfd8-42d6-8d37-28c92ecb63b0",
+   "metadata": {},
+   "outputs": [
```
Comment (atheo89) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb:8]: We don't need the `outputs` metadata. 
To remove them, once you run the notebook and all tests passed, restart the Kernel of the notebook, then copy-paste it into your test_notebook.ipynb

File: jupyter/trustyai/ubi8-python-3.8/test/test_notebooks.ipynb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    "\n",
+    "suite = unittest.TestLoader().loadTestsFromTestCase(TestTrustyaiNotebook)\n",
+    "unittest.TextTestRunner().run(suite)\n",
+    "\n"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3.9",
```
Comment (atheo89) [inline jupyter/trustyai/ubi8-python-3.8/test/test_notebooks.ipynb:69]: This notebook is for ubi8 flavor, so it needs to be changed to 3.8.
To do that you need to run the contents of the notebook on a ubi8 flavor (In order to be corrected automatically)

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thank for the work
can you please squash all the commit to 1 please.

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb [hunk 5]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (rkpattnaik780) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebooks.ipynb:1]: nit: can we rename the test notebooks to `test_notebook.ipynb` to keep it consistent with naming of other test notebooks we have.

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebook.ipynb [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+     "data": {
+      "text/plain": [
+       "<unittest.runner.TextTestResult run=5 errors=3 failures=0>"
+      ]
+     },
+     "execution_count": 37,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
```
Comment (jstourac) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebook.ipynb:54]: I don't understand this deeply yet, but suppose that we can left the outputs field empty?

```suggestion
   "outputs": [],
```

**Review by jstourac — COMMENTED** [review]
One coment, otherwise LGTM.

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebook.ipynb [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    "    def test_python_version(self): \n",
+    "        expected_major_minor = '3.9' # Set the expected version (x.y) \n",
+    "        actual_major_minor = '.'.join(python_version().split('.')[:2])\n",
+    "        self.assertEqual(actual_major_minor, expected_major_minor, \"incorrect version\")\n",
+    "\n",
+    "    def test_trustyai_version(self): \n",
+    "        expected_major_minor = '0.2' # Set the expected version (x.y) \n",
+    "        actual_major_minor = '.'.join(trustyai.__version__.split('.')[:2]) \n",
+    "        self.assertEqual(actual_major_minor, expected_major_minor, \"incorrect version\")\n",
+    "     def test_fairnessmetrics(self):\n",
```
Comment (jstourac) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebook.ipynb:76]: Indentation:

```suggestion
    "    def test_fairnessmetrics(self):\n",
```

**Review by jstourac — CHANGES_REQUESTED** [review]

**Comment by atheo89** [comment]
Upon reviewing the test notebook for the Ubi9, it appears to be distinct from the one I examined during my previous review.  Could you please revert it? 
It should look like the ubi8 -> https://github.com/dibryant/notebooks/blob/trustyai/jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb

File: jupyter/trustyai/ubi9-python-3.9/test/test_notebook.ipynb [hunk 8]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (atheo89) [inline jupyter/trustyai/ubi9-python-3.9/test/test_notebook.ipynb:1]: Upon reviewing the test notebook for the Ubi9, it appears to be distinct from the one I examined during my previous review. Could you please revert it?
It should look like the ubi8 -> https://github.com/dibryant/notebooks/blob/trustyai/jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb

**Review by atheo89 — CHANGES_REQUESTED** [review]
Some changes are required

**Comment by atheo89** [comment]
The e2e is normal that it failed on TrustyAI step this is because is not yet merged into the main -> https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/opendatahub-io_notebooks/244/pull-ci-opendatahub-io-notebooks-main-notebooks-e2e-tests/1714189236862717952

**Pull request merged**